sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/math.scala

```diff
@@ -51,16 +54,28 @@ abstract class LeafMathExpression(c: Double, name: String)
  * @param f The math function.
  * @param name The short name of the function
  */
-abstract class UnaryMathExpression(f: Double => Double, name: String)
+abstract class AbstractUnaryMathExpression[T, U](name: String)
```

Comment:
but we still list it in math functions? It might look inconsistent? Or should we create another abstract class like `UnaryGeneralMathExpression` based on `UnaryExpression` alongside current `UnaryMathExpression`?

Or just like you said, we directly let `BIN` inherit `UnaryExpression` and `ExpectsInputTypes`?